### ISSUE TITLE
Slice bounds out of range in proc.newCompositeMemory

### ISSUE THREAD

**Issue by thallgren** [derekparker/delve#1416]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
Tried both:
Delve Debugger
Version: 1.1.0
Build: $Id: 1990ba12450cab9425a2ae62e6ab988725023d5c $
and
Build: 3cfb00b2cc0edc2b32bf90b30c64724fb4acc99c

2. What version of Go are you using? (`go version`)?
Tried both:
go version go1.11.2 linux/amd64
and
go version devel +310f2e2465 Sun Nov 18 17:00:40 2018 +0000 linux/amd64

3. What operating system and processor architecture are you using?
GNU/Linux x86_64 

4. What did you do?
Set breakpoint from Goland IDE, debug tests, and hit breakpoint.

5. What did you expect to see?
A stack backtrace report.

6. What did you see instead?
A panic from github.com/derekparker/delve/pkg/proc.newCompositeMemory. Full stack trace:
```
org.jetbrains.concurrency.InternalPromiseUtil$MessageError: Internal debugger error: runtime error: slice bounds out of range
runtime.call32 (0x457c2a)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/runtime/asm_amd64.s:522
runtime.gopanic (0x42c078)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/runtime/panic.go:513
runtime.panicslice (0x42afe9)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/runtime/panic.go:54
github.com/derekparker/delve/pkg/proc.newCompositeMemory (0x6ac68c)
	/mnt/agent/work/e7ce1e3d82fa1892/pkg/proc/mem.go:105
github.com/derekparker/delve/pkg/proc.(*EvalScope).extractVarInfoFromEntry (0x6cd778)
	/mnt/agent/work/e7ce1e3d82fa1892/pkg/proc/variables.go:911
github.com/derekparker/delve/pkg/proc.(*EvalScope).Locals (0x6d5576)
	/mnt/agent/work/e7ce1e3d82fa1892/pkg/proc/variables.go:2100
github.com/derekparker/delve/pkg/proc.(*EvalScope).LocalVariables (0x6cb45e)
	/mnt/agent/work/e7ce1e3d82fa1892/pkg/proc/variables.go:685
github.com/derekparker/delve/service/debugger.(*Debugger).convertStacktrace (0x72d90e)
	/mnt/agent/work/e7ce1e3d82fa1892/service/debugger/debugger.go:974
github.com/derekparker/delve/service/debugger.(*Debugger).Stacktrace (0x72d273)
	/mnt/agent/work/e7ce1e3d82fa1892/service/debugger/debugger.go:952
github.com/derekparker/delve/service/rpc2.(*RPCServer).Stacktrace (0x8cc5a8)
	/mnt/agent/work/e7ce1e3d82fa1892/service/rpc2/server.go:176
runtime.call64 (0x457caa)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/runtime/asm_amd64.s:523
reflect.Value.call (0x4b24d8)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/reflect/value.go:447
reflect.Value.Call (0x4b1f73)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/reflect/value.go:308
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).serveJSONCodec.func2 (0x8d4dcc)
	/mnt/agent/work/e7ce1e3d82fa1892/service/rpccommon/server.go:321
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).serveJSONCodec (0x8d32eb)
	/mnt/agent/work/e7ce1e3d82fa1892/service/rpccommon/server.go:323
runtime.goexit (0x459940)
	/mnt/agent/work/e7ce1e3d82fa1892/go/src/runtime/asm_amd64.s:1333
```

**Comment by thallgren** [comment]
Logs after stopping at breakpoint with https://github.com/derekparker/delve/pull/1417 applied. A  seemingly correct stack trace is now shown in the "Frames" window.
[logs-20181119-144454.zip](https://github.com/derekparker/delve/files/2595511/logs-20181119-144454.zip)

**Comment by aarzilli** [comment]
Ok, as far as delve is concerned #1417 is good enough.

cc @heschik I think this could be a problem with the location lists.

The logs are a bit verbose (to say the least) so I'll give an abridged version, compile this:

```
package main

import "fmt"

func ExampleSomething() {
	fmt.Printf("blah\n")
}
```

using `go test -c` (no gcflags, because of the bug in 1.11 they don't get applied anyway in this case).
Look at local variable `err` of `testing.runExamples`, it has a location list containing an expression that goes like this: `DW_OP_reg2 DW_OP_piece 0x8 DW_OP_reg0 DW_OP_piece 0x8`. The problem is that its range also covers a CALL instruction (specifically to `testing.runExample`).

1. my interpretation of the standard is that registers only exist for frames other than the topmost if the FDE says where they are (am I wrong?)
2. registers are all caller saved in go anyway

I don't think this is an important bug, just wanted to let you know.

**Comment by heschik** [comment]
I didn't look too deeply, but glancing at the location list debug output, I think this is an instance of the problem fixed by https://golang.org/cl/146718, which extends location lists entries across places they aren't valid. @dr2chase, FYI. We should maybe double-check once that CL is in.

Total tangent: do you need more attention on the gcflags bug? I only noticed it a couple days ago and it seems like we want it fixed for 1.12...

**Comment by aarzilli** [comment]
>Total tangent: do you need more attention on the gcflags bug? I only noticed it a couple days ago and it seems like we want it fixed for 1.12...

I was going to lobby for this next week at the meeting, actually.

**Comment by aarzilli** [comment]
I think we can consider this solved on our side by 7f53117ea94b51413efadfe3d7a4e34615d5ef2b
